packages/thirdweb/src/indexer/endpoints/getBlock.ts

```diff
@@ -0,0 +1,77 @@
+import type { BlockTag, GetBlockReturnType } from "viem";
+import type { Chain } from "../../chains/types.js";
+import type { ThirdwebClient } from "../../client/client.js";
+import { getClientFetch } from "../../utils/fetch.js";
+import { formatIndexerBlock } from "../formatter.js";
+import type { IndexerInternalBlock, IndexerResponse } from "../types.ts";
+import { getBlockEndpoint } from "../urls.js";
+
+export type GetBlockParams = {
+  /**
+   * A client is the entry point to the thirdweb SDK. It is required for all other actions.
+   *
+   * You can create a client using the `createThirdwebClient` function.
+   * Refer to the [Creating a Client](https://portal.thirdweb.com/typescript/v5/client) documentation for more information.
+   *
+   */
+  client: ThirdwebClient;
+  /**
+   * Number of the block to fetch
+   */
+  blockNumber: bigint;
+  /**
+   * Chain the block can be found on
+   */
+  chain: Chain;
+};
+
+export type GetBlockResult<TBlockTag extends BlockTag = "latest"> = {
+  block: GetBlockReturnType<undefined, false, TBlockTag>;
+};
+
+/**
+ * @beta
+ *
+ * Get data for a single block
+ *
+ * @param {GetBlockParams} params
+ * @returns {Promise<GetBlockResult>}
+ *
+ * @example
+ * ```ts
+ * import { createThirdwebClient, getBlock, defineChain } from "thirdweb";
+ *
+ * const client = createThirdwebClient({ clientId: "..." });
+ * const { block } = await getBlock({
+ *  client,
+ *  blockNumber: 9662167n,
+ *  chain: defineChain(1)
+ * });
+ * ```
+ */
+export async function getBlock(
+  params: GetBlockParams,
+): Promise<GetBlockResult> {
```

Comment:
I'd argue this should just return `block` directly not an object with `block` in it?